benchopt/benchmark.py

```diff
@@ -201,17 +201,36 @@ def install_all_requirements(self, include_solvers, include_datasets,
         """
         # Collect all classes matching one of the patterns
         print("Collecting packages:")
+
+        install_solvers = True
+        install_datasets = True
+
+        # If -d is used but not -s, then does not install any solver
+        if len(include_solvers) == 0 and len(include_datasets) > 0:
+            install_solvers = False
+
+        # If -s is used but not -d, then does not install any dataset
+        if len(include_datasets) == 0 and len(include_solvers) > 0:
+            install_datasets = False
+
+        # If -d or -s are followed by 'all' then all
+        # solvers or datasets are included
+        if 'all' in include_solvers:
+            include_solvers = []
+        if 'all' in include_datasets:
+            include_datasets = []
+
         conda_reqs, shell_install_scripts, post_install_hooks = [], [], []
         check_installs = []
-        for list_classes, include_patterns in [
-                (self.get_solvers(), include_solvers),
-                (self.get_datasets(), include_datasets)
+        for list_classes, include_patterns, install_pattern in [
```

Comment:
nitpick: from the variable name I expected it to be a regexp pattern or similar, but it's a bool. Can we come up with a name reflecting that this determines whether or not we will install stuff ?

Reply:
What do you think of install_flag, install_required, is_required ?

Reply:
or `to_install` ?

Reply:
ok